pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/SingleConnectionBrokerRequestHandler.java

```diff
@@ -90,6 +90,8 @@ protected BrokerResponse processBrokerRequest(long requestId, BrokerRequest orig
     Map<ServerRoutingInstance, ServerResponse> response = asyncQueryResponse.getResponse();
     _brokerMetrics
         .addPhaseTiming(rawTableName, BrokerQueryPhase.SCATTER_GATHER, System.nanoTime() - scatterGatherStartTimeNs);
+    _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.REQUEST_SIZE,
+        originalBrokerRequest.toString().length());
```

Comment:
Calling toString() here is not a good idea for high throughput use cases. See if you can get this value from the netty layer.

Reply:
I am not sure if we can get user's original query at netty layer(query has been converted as thrift object at netty).